servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java

```diff
@@ -701,6 +707,10 @@ private static Pattern compileCharsetRegex(String charsetName) {
         return compile(".*;\\s*charset=\"?" + quote(charsetName) + "\"?\\s*(;.*|$)", CASE_INSENSITIVE);
     }
 
+    private static boolean hasCharset(final CharSequence contentTypeHeader) {
+        return contentTypeHeader.toString().contains("charset=");
```

Comment:
Good catch. I'll fix it and add this test.

> where CHARSET_PATTERN = compile(".+charset=.+", CASE_INSENSITIVE)

Probably we should use `".+;\\s*charset=.+"`, in case the header contains unknown fields that end with *charset?